enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,372 @@
+kubernetes controllers that create resources via cloud service provider api and other open source tools. A reconciliation 
+using a generic metadata controller helps to keep the tags synchronized for cloud resources. 
+
+In the following proposal, the controller supports mainly 
+1. Day 2 operations and tag management for cloud resources selected using classifiers.
+2. Own the tag list and reconcile to replace any edits performed external to the controller.
+3. Opcodes in specification for handling user request as an operation on tag list.
+4. Extensible APIs and segregated configuration for controller and cloud provider specific configurations.
+
+## Motivation
+
+Users should be able to maintain lifecycle of tags for cloud resources created during installation or by other day-2 kubernetes
+controllers and operators using kubernetes API.
+
+### User stories
+
+As a cluster admin, I want to add tags to the cloud resources created at cluster installation,  which help
+for cost calculation and reports.
+
+As a cluster admin, I want tags to be added to the cloud resources created post cluster installation which help to include
+cloud resources for other tag processors.
+
+As a cluster admin, I want all tags to be added via metadata CR, which helps to reconcile the tag key and values.
+
+As a cluster admin, I want to delete tags, which are not required on the cloud resources.
+
+As a cluster admin, I want to add tags on selected resources, which are identified using specified labels/classifiers.
+
+As a cluster admin, I want to be able to ignore cloud resource from all tag operations, which helps to avoid overwriting values 
+set by cloud service provider's policies.
+
+As a cluster admin, I want the tag list to be lexicographically sorted, to have a deterministic set of tags applied 
+when tag list size exceeds the maximum allowed limit on cloud resource.
+
+As a cluster admin, I want the tag list to be lexicographically sorted, to have a deterministic set of tags appended to 
+existing tags for cloud resources when the tag list exceeds the maximum allowed limit on cloud resource.
+
+As a cluster admin, I want to use a CLI, that helps to securely login to cloud service provider and perform configuration and management operations for tags.
+
+### Goals
+
+1. Enable continuous management operations (create, append , update and delete) applicable for tagging 
+ cloud resources created by kubernetes services.
+2. Provide a new custom resource of kubernetes api for specification and status of tags.
+3. Enable continuous monitoring of tags applied and reconcile when there are changes made external to 
+the controller's API.
+4. Provide CLI for operations, cloud service provide authentication and ability to edit custom resource with ease.
+
+### Non-goals
+
+1. Managing metadata which are not related to cloud resource tagging.
+2. Data-related metadata for stored objects and records, for example, S3 bucket vs S3 object.
```

Comment:
can you make this clearer?  I think it's trying to say we'd manage the metadata for things like s3 buckets, but not for things like s3 objects.  (we'd manage the metadata for the cloud resource but not the content within the resource)

i'd also say, assuming i got that right, that it's more than a non-goal.  Non-goals are things you aren't setting out to solve, but take no opinion on whether they happen or not.  But we already know managing metadata on s3 objects is highly problematic because of the tag limit, right?  So i think we should explicitly state that those resource types are off the table for being managed by this controller.

(also if there's another reason to avoid managing metadata on those objects, beyond just the problem of the tag limit constraints, we should state that somewhere.)